drivers/auxdisplay/tm1637.c

```diff
@@ -475,17 +479,19 @@ static ssize_t tm1637_store_led(struct device *dev,
 	mutex_lock(&tm->lock);
 
 	if (kstrtoul(buf, 0, &data))
-		return -EINVAL;
+		ret = -EINVAL;
+		goto unlock;
```

Comment:
You've not tested this - this `goto` is unconditional.

But you're wasting your time - as I said previously:
> The biggest problem is that the driver should be sent upstream to the Linux kernel maintainers.

Reply:
No - not sysfs, libgpiod which is ioctl-based and much quicker. How fast does one need to update a 4x7-segment display?